### PR TITLE
EOS-17946:HA event analyzers alert parser

### PR DESCRIPTION
**Pull request by VijayThakkar132**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-17946
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
   Implement alert parser which takes incoming alerts as input and returns event object.
  </code>
</pre>
## Solution
<pre>
  <code>
    Implemented alert parser.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Added integration test for the validation.
  </code>
</pre>

### CODE REVIEW

File: ha/core/event_analyzer/parser/parser.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        event = {
+            "event_id" : alert.get('sensor_response_type').get('alert_id'),
+            "event_type" : alert.get('sensor_response_type').get('alert_type'),
+            "severity" : alert.get('sensor_response_type').get('severity'),
+            "site_id" : alert.get('sensor_response_type').get('info').get('site_id'),
+            "rack_id" : alert.get('sensor_response_type').get('info').get('rack_id'),
+            "cluster_id" : alert.get('sensor_response_type').get('info').get('cluster_id'),
+            "storageset_id" : "TBD",
+            "node_id" : alert.get('sensor_response_type').get('info').get('node_id'),
```
Comment (ajay-paratmandali) [inline ha/core/event_analyzer/parser/parser.py:57]: Handle sensor_response_type and actureator_responce_type

File: ha/core/event_analyzer/parser/parser.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    def parse_event(self, msg: str) -> HealthEvent:
+        """
+        Parse event.
+        Args:
+            msg (str): Msg
+        """
+        alert = json.loads(msg)
+
+        event = {
+            "event_id" : alert.get('sensor_response_type').get('alert_id'),
```
Comment (ajay-paratmandali) [inline ha/core/event_analyzer/parser/parser.py:50]: What happen if msg format is one of this field

can we somehow caught and log this exception and raise error

Comment (indrajitzagade) [reply ha/core/event_analyzer/parser/parser.py:50]: Let's add try-catch block for this whole parsing code here?

File: ha/core/event_analyzer/parser/parser.py (hunk 1 above)
Comment (VijayThakkar132) [reply ha/core/event_analyzer/parser/parser.py]: Does not required handling for the actuator response type as We already filter out alerts with actuator response type in the filter_event() method.

File: ha/core/event_analyzer/parser/parser.py (hunk 2 above)
Comment (VijayThakkar132) [reply ha/core/event_analyzer/parser/parser.py]: I have removed .get() method and used [] to access the key of the dictionary as .get() method does not raise exception and returns None if the key is not present in the dictionary.
Added try-catch block for the exception handling.

File: ha/test/integration/event_analyzer/test_alert_parser.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import pathlib
+
+# Test case for alert filter
+if __name__ == '__main__':
+    sys.path.append(os.path.join(os.path.dirname(pathlib.Path(__file__)), '..', '..', '..', '..'))
+    from ha.core.event_analyzer.parser.parser import AlertParser
+    from ha.core.system_health.model.health_event import HealthEvent
+    import traceback
+    try:
+        alert_parser = AlertParser()
```
Comment (ajay-paratmandali) [inline ha/test/integration/event_analyzer/test_alert_parser.py:30]: Please add negative case also
to get exception

File: ha/test/integration/event_analyzer/test_alert_parser.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    "location": "Enclosure 0 - Right",
+                    "position": "Indexed",
+                    "health-recommendation": "Install the missing fan module."
+                },
+                "alert_id": "15740759091a4e14bca51d46908ac3e9102605d560",
+                "host_id": "s3node-host-1"
+            }
+        }
+
+        health_event = alert_parser.parse_event(json.dumps(TestMsg))
```
Comment (ajay-paratmandali) [inline ha/test/integration/event_analyzer/test_alert_parser.py:69]: Add negative case also

```
del TestMsg["alert_id"]
try:
      health_event = alert_parser.parse_event(json.dumps(TestMsg))
except Exception as e:
      print(f"Negative case is successful, caught err: {e}")
```

Comment (VijayThakkar132) [reply ha/test/integration/event_analyzer/test_alert_parser.py]: Done

**Review by ajay-paratmandali — APPROVED** [review]

File: ha/core/event_analyzer/parser/parser.py (hunk 2 above)
Comment (indrajitzagade) [reply ha/core/event_analyzer/parser/parser.py]: ok

**Review by indrajitzagade — APPROVED** [review]

**Review by azheregelya — APPROVED** [review]

**Pull request merged**